subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractIntegrationSpec.groovy

```diff
@@ -95,10 +95,12 @@ class AbstractIntegrationSpec extends Specification {
     def setup() {
         m2.isolateMavenLocalRepo(executer)
         executer.beforeExecute {
-            executer.withArgument("-Dorg.gradle.internal.repository.max.tentatives=$maxHttpRetries")
+            withArgument("-Dorg.gradle.internal.repository.max.tentatives=$maxHttpRetries")
             if (maxUploadAttempts != null) {
-                executer.withArgument("-Dorg.gradle.internal.network.retry.max.attempts=$maxUploadAttempts")
+                withArgument("-Dorg.gradle.internal.network.retry.max.attempts=$maxUploadAttempts")
             }
+            withArgument("-Porg.gradle.java.installations.auto-detect=false")
```

Comment:
I think it would be better to have a boolean in `AbstractGradleExecuter` which disables toolchain auto detect and auto download. It could be something like `disableToolchainDownloadAndDetection` which would be `true` by default. When `true`, `getAllArgs()` would add the two flags as gradle properties (with `false`), when `false`, `getAllArgs()` would add no flags.

This way, tests which test the toolchain detection or auto download can pass the proper flags to the daemon, while all other tests disable detection and auto-download. It may make sense to introduce two boolean flags on the executor as well, if it makes to set the detection and auto download flag different in tests.

The important part is here that there is way to not pass any properties with regards to detection/auto-download to the daemon and the test can the explicitly set what it wants to test.

Reply:
Good call. Was convinced at first but it looks a lot better now with the fluent API on the executor.